.github/workflows/sonar.yml

```diff
@@ -0,0 +1,30 @@
+name: SonarCloud
+
```

Comment:
To enable CI for PRs from forked repos, I think we should use the same approach as used in the snap: https://github.com/canonical/prometheus-juju-exporter/pull/6. I believe this approach should be adopted by all of our GitHub projects.